Specs/Core/loadWithXhrSpec.js

```diff
@@ -133,18 +143,364 @@ defineSuite([
             });
         });
 
-        xit('can support 2xx HTTP status (other than 200)', function(){
+        xit('can support 2xx HTTP status (other than 200)', function() {
             return loadWithXhr({
-                method: 'POST',
-                url: 'http://jsonplaceholder.typicode.com/posts',
-                data: {
-                    title: 'foo',
-                    body: 'bar',
-                    userId: 1
+                method : 'POST',
+                url : 'http://jsonplaceholder.typicode.com/posts',
+                data : {
+                    title : 'foo',
+                    body : 'bar',
+                    userId : 1
                 }
-            }).then(function(result){
+            }).then(function(result) {
                 expect(JSON.parse(result).id).toEqual(101);
             });
         });
     });
+
+    describe('URL loading', function() {
+        var fakeXHR;
+
+        beforeEach(function() {
+            fakeXHR = jasmine.createSpyObj('XMLHttpRequest', ['send', 'open', 'setRequestHeader', 'abort', 'getAllResponseHeaders']);
```

Comment:
Creating a completely fake spy object here (at least partially) invalidates the entire test.  The crux of the problem is that XMLHttpRequest has different behaviors in different browsers, so mocking it to be consistent in all cases isn't actually testing what we need to test.